<a href="https://colab.research.google.com/github/JosenildoJunior/StatPyDataScience/blob/main/Probabilidade_estat%C3%ADstica.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Estatística com Python: Um Guia para Estudos e Solução de Problemas**


## Breve Resumo:

Probabilidade é a área da matemática que estuda a chance de ocorrência de eventos.

# **Permutação**

Breve Resumo:

Permutação é uma técnica de contagem utilizada para determinar quantas maneiras existem para ordenar os elementos de um conjunto finito.

Cálculo:

O número de permutações de um conjunto com n elementos é dado pela seguinte fórmula:

- P(n) = n!

 Realizando importações necessarias

In [2]:
# Manipulação de arquivos
import pandas as pd

# Algébra linear
import numpy as np

# Funções estatísticas
import scipy.stats

# Funções matemáticas
import math

# Visualização de dados
import seaborn as sns
import matplotlib.pyplot as plt

Digamos que tenhamos a seguinte pergunta:
- Dados 5 números, calcule o número de permutações possíveis.

Para resolvermos essa questão, basta usar a função factorial() do pacote math. Como sabemos que é uma permutação de 5, basta aplicar a função para sabermos o valor de 5!

In [5]:
# Calculando o fatorial de 5
math.factorial(5)

120

Portanto, podemos afirmar que nesse caso existem 120 permutações possíveis utilizando apenas 5 números

Resolva essa questâo:

- Ao criar uma senha com 5 dígitos, utilizando qualquer letra e números de 0 a 9 e levando em consideração que as letras e números **não podem se repetir**, quantas permutações são possíveis?

**Esse é um caso de permutação de subconjunto onde utilizaremos a seguinte formula:**
- $P(n, k) = n! / (n-k)!$

In [6]:
# Aplicando a fórmula da permutação de subconjunto
math.factorial(36) / math.factorial(36 -5)

45239040.0

Assim, podemos observar que chegamos ao valor de 452 milhões de possibilidades diferentes

Resolva essa questâo:
- Ao criar uma senha com 5 dígitos, utilizando qualquer letra e números de 0 a 9 e levando em consideração que as letras e números **podem se repetir**, quantas permutações são possíveis?

Como os valores podem se repetir, aplicaremos uma abordagem um pouco diferente. Utilizaremos a seguinte fórmula:

- $P(n, k) = n^k$

Agora, basta aplicar a fórmula nos valores para aplicar a permutação

In [7]:
# Aplicando a fórmula
math.pow(36, 5)

60466176.0

Dito isso, podemos montar 60 milhões de senhas de diferentes fórmulas, podendo utilizar valores diferentes.